devito/operations/interpolators.py

```diff
@@ -388,3 +389,520 @@ def callback():
+            option = 2, pontos extremos da vizinhança.
+            Onde 1 < |arg| < 2
+            """
+            a = -1/2
+            if option == 1:
+                return (a+2)*(abs(arg)**3) - (a+3)*(abs(arg)**2) + 1
+            elif option == 2:
+                return a*(abs(arg)**3) - 5*a*(abs(arg)**2) + 8*a*abs(arg) - 4*a
+
+        # Values ​​that define which cubic kernel will be used
+        option = [2, 1, 1, 2]
+
+        pos = sympy.Matrix(position)
+
+        dim_pos = sympy.Matrix(dim_pos)
+
+        # Neighboring points(interpolation) or source value (injection)
+        p = [expr.xreplace(subs) for subs in idx_subs]
+        p = sympy.Matrix(p)
+
+        # Defines the kernel witch will be multiplied by the neighboring
+        # points(or source value)
+        base_index = [idx3d, idx2d]
+        index_opt = [b for b in base_index if b is not None]
+        arg_kernel = [pp - dim for pp, dim in zip(pos, dim_pos)]
+
+        kernel = [cubicConv(arg, option[opt]) for arg, opt in zip(arg_kernel, index_opt)]
+        kernel = np.prod(kernel)
+        kernel = [kernel*cubicConv(arg_kernel[-1], option[ii]) for ii in range(4)]
+
+        # Defining the final shape of the cubic equation
+        points = [point*k.subs(idx) for point, k, idx in zip(p, kernel, idx_subs)]
+
+        return points
+
+    def _bicubic_equations(self, expr, idx_subs, dim_pos=None):
+        """
+        Generate equations responsible for bicubic interpolation's computation.
+        Parameters
+        ----------
+        expr : expr-like
+            Input expression to interpolate.
+        idx_subs: dict
+            Structure responsible for mapping the order of substitution
+            of dimensions in expr.
+        dim_pos: tuple
+            Structure containing the spatial dimensions of the field
+        """
+        pos = self.sfunction._point_symbols
+
+        eqs = [self._cubic_equation(expr, pos, dim_pos,
+               idx_subs=idx_subs[ii*4:(ii+1)*4], idx2d=ii)
+               for ii in range(4)]
+        return eqs
+
+    def _tricubic_equations(self, expr, idx_subs, dim_pos=None):
+        """
+        Generate equations responsible for tricubic interpolation's computation.
+        Parameters
+        ----------
+        expr : expr-like
+            Input expression to interpolate.
+        idx_subs: dict
+            Structure responsible for mapping the order of substitution
+            of dimensions in expr.
+        dim_pos: tuple
+            Structure containing the spatial dimensions of the field
+        """
+        pos = self.sfunction._point_symbols
+
+        eqs = []
+        eqs.extend([self._cubic_equation(expr, pos, dim_pos,
+                    idx_subs=idx_subs[(ind*4)+(16*ii):((1+ind)*4)+16*ii],
+                    idx2d=ind, idx3d=ii)
+                    for ii in range(4) for ind in range(4)])
+
+        return eqs
+
+    def _interpolation_indices(self, variables, offset=0, field_offset=0):
```

Comment:
isn't there anything we could factorize in a common superclass?

Reply:
> isn't there anything we could factorize in a common superclass?

Yes, I think it has. I'm working on it

Reply:
I updated the code by adding the superclass you suggested